assets/javascripts/discourse/templates/components/chat-channel-row.hbs

```diff
@@ -11,9 +11,14 @@
   {{chat-channel-title channel=channel unreadIndicator=true}}
 
   {{#if options.leaveButton}}
-    {{chat-channel-leave-btn
-      channel=channel
-      onLeaveChannel=(action "onLeaveChannel")
-    }}
+    {{#unless site.mobileView}}
+      <ChatToggleMembershipBtn
+        @channel={{channel}}
+        @afterToggle={{action "afterMembershipToggle"}}
+        @leaveIcon="times"
+        @leaveClasses="btn-flat chat-channel-leave-btn"
```

Comment:
`chat-channel-leave-btn` is very loaded and styled in a lot of CSS files which is why I left it in, not sure if it would be better to move all those styles to use `chat-toggle-membership-btn__leave-channel` ?

Reply:
I think we could do

```
toggle-channel-membership-btn -leave
toggle-channel-membership-btn -join
```

That would be right @chapoi no?

Reply:
I thought it would be `__leave` since it's an element under the block? Vs `--leave` if it was a modifier, like `--danger`?

Reply:
I mainly wasn't sure whether I should change the CSS files to no longer use `chat-channel-leave-btn` and to target this new class

Reply:
I know @chapoi has been using this in browse page for -closed and -archived so I would think it can be used here too?

Reply:
Sorry, I don't get mails from these mentions, should change that
I use modifiers modified (get it) with just 1 dash. So indeed -leave